### PR TITLE
Add Buffer.alloc to nodejs externs (fixes #3263)

### LINKED ISSUE #3263: contrib/nodejs/buffer.js is outdated
This 

https://github.com/google/closure-compiler/blob/master/contrib/nodejs/buffer.js 

is outdated wrt to 

https://npmdoc.github.io/node-npmdoc-buffer/build/apidoc.html#apidoc.element.buffer.Buffer.alloc

resulting in `JavaScript exception: TypeError: Buffer.nb is not a function` error (`JavaScript exception: TypeError: Buffer.$alloc$ is not a function` when compiled with `--debug`) in my code. The following addition (a partial hack) fixed it for me:

```
/**
 * @param {number=} size
 * @return {Buffer}
 */
buffer.Buffer.alloc;
```

Note that I get the same error when using https://github.com/dcodeIO/node.js-closure-compiler-externs/blob/master/buffer.js, which is more complete, but even older and so similarly outdated, so instead I report here.

### PR DESCRIPTION
**Pull request by Mikolaj**
See #3263.

### CODE REVIEW

**Comment by Mikolaj** [comment]
I signed it.

File: contrib/nodejs/buffer.js [hunk 1]
```diff
@@ -55,3 +55,11 @@ buffer.SlowBuffer.prototype.slice;
  * @return {string}
  */
 buffer.SlowBuffer.prototype.toString;
+
+/**
+ * @param {number} size
+ * @param {(string|Buffer|number)=} fill
```
Comment (tjgq) [inline contrib/nodejs/buffer.js]: Write `!Buffer` to make it non-nullable.

File: contrib/nodejs/buffer.js [hunk 2]
```diff
@@ -55,3 +55,11 @@ buffer.SlowBuffer.prototype.slice;
...
+ * @param {number=} encoding
```
Comment (tjgq) [inline contrib/nodejs/buffer.js]: According to https://nodejs.org/api/buffer.html#buffer_class_method_buffer_alloc_size_fill_encoding, I think this should be `string=`?

File: contrib/nodejs/buffer.js [hunk 3]
```diff
@@ -55,3 +55,11 @@ buffer.SlowBuffer.prototype.slice;
...
+ * @return {Buffer}
```
Comment (tjgq) [inline contrib/nodejs/buffer.js]: Write `!Buffer` to make it non-nullable.

File: contrib/nodejs/buffer.js (hunk 2 above)
Comment (Mikolaj) [reply contrib/nodejs/buffer.js]: Oh, yes, good catch.

File: contrib/nodejs/buffer.js (hunk 1 above)
Comment (Mikolaj) [reply contrib/nodejs/buffer.js]: Will do.

File: contrib/nodejs/buffer.js (hunk 3 above)
Comment (Mikolaj) [reply contrib/nodejs/buffer.js]: Done. Let's see if this fixes travis.

**Comment by tjgq** [comment]
Created Google internal issue b/128616787. Imported for internal review and testing.

Let's ignore the Travis failure for now - I believe it's related to https://github.com/google/closure-compiler-npm/issues/130 which @ChadKillingsworth is trying to fix.

**Pull request closed without merging**